core/src/main/java/org/mskcc/cbio/portal/dao/DaoGeneOptimized.java

```diff
@@ -221,6 +225,60 @@ private void cacheGene(CanonicalGene gene) {
     public static DaoGeneOptimized getInstance() {
         return daoGeneOptimized;
     }
+    
+    /**
+     * Return geneticEntityId from cache for given entrezGeneId
+     * 
+     * @param entrezGeneId
+     * @return
+     */
+    public static int getGeneticEntityId(long entrezGeneId) {
+    	//get entity id from cache:
+    	CanonicalGene gene = daoGeneOptimized.getGene(entrezGeneId);
+		if (gene != null) {
+			return gene.getGeneticEntityId();
+		}
+		else {
+			throw new RuntimeException("Invalid entrezGeneId symbol. Not found in cache: " + entrezGeneId);
+		}    			
+    }
+    
+    public static int getGeneticEntityId(long entrezGeneId, String hugoGeneSymbol) {
```

Comment:
If caller passes in both entrezGeneId *and* hugoGeneSymbol, then hugoGeneSymbol is ignored. With this setup, it is possible that entrezGeneId and hugoGeneSymbol arguments will refer to different genes, and this function will not notice the error. One improvement would be to throw an exception if both arguments are not "empty". Another approach would be to resolve the geneticEntityId for both arguments and then throw an exception if the geneticEntityIds for the two arguments differ. We could also have two separate functions ... one which takes entrezGeneId only, and one which takes hugoGeneSymbol only.
None of these changes are required ... I'm just raising the issue for possible consideration.